kotlin/internal/jvm/compile.bzl

```diff
@@ -67,6 +67,81 @@ def _fold_jars_action(ctx, rule_kind, output_jar, input_jars, action_type = ""):
         ),
     )
 
+#
+# Creates an action to build a source jar.
+# Source jars can be built using ./bazel build //foo/bar:bar-sources.jar
+#
+def _src_jar_action(ctx, output_jar):
+    args = ctx.actions.args()
+    args.add_all([
+        "--normalize",
+        "--compression",
+        "--exclude_build_data",
+        "--warn_duplicate_resources"
+    ])
+    args.add("--output", output_jar);
+
+    args.add_all("--resources", ctx.files.srcs);
+
+    ctx.actions.run(
+        mnemonic = "KotlinSrcJar",
+        inputs = ctx.files.srcs,
+        outputs = [output_jar],
+        executable = ctx.executable._singlejar,
```

Comment:
We can pull singlejar off the java toolchain, or add it to the kotlin one. As long as we don't make a loop, I favor pushing tools to the toolchain.

Reply:
Done.

Discovered `java_common.pack_sources()` so using this to create the source jar. For the existing Fold Jars actions getting singlejar from the java toolchain now.